var/spack/repos/builtin/packages/hdf-eos5/package.py

```diff
@@ -77,15 +78,23 @@ def url_for_version(self, version):
                 "version/checksum not found in version_list".format(version)
             )
 
+    @run_before("configure")
+    def fix_configure(self):
+        # spack patches the configure file unless autoconf is run,
+        # and this fails because configure has the wrong permissions (644)
+        if not self.force_autoreconf:
+            chmod(join_path(self.stage.source_path, "configure"), 0o755)
+
+        # The configure script as written really wants you to use h5cc. This causes
+        # problems because h5cc differs when HDF5 is built with autotools vs cmake,
+        # and we lose all nice flags from the Spack wrappers. These filter operations
+        # allow use to use the Spack wrappers again.
+        filter_file("\\$CC -show &> /dev/null", "true", "configure")
+        filter_file("CC=./\\$SZIP_CC", "", "configure")
```

Comment:
You might fine the following useful:
1. [`keep_modification_time`](https://github.com/spack/spack/blob/5b6137d91ada0f7b072094b6872277bec7830bf2/lib/spack/llnl/util/filesystem.py#L2782), which is usually used when patching files generated by Autotools (to avoid the following re-generation at the build time due to inconsistent timestamps).
2. [`string`](https://github.com/spack/spack/blob/5b6137d91ada0f7b072094b6872277bec7830bf2/lib/spack/llnl/util/filesystem.py#L322) argument of `filter_file`.
3. Python raw strings, i.e. `r"whatever"`, which are often used for the regular expression.